### PR TITLE
Anonymize user

### PR DESCRIPTION
**Pull request by MrHanSan**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [ ] I have added tests for the code I added
- [ ] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

- [ ] I have updated the build configuration
- [ ] These changes requires changes to configuration in production 
    - [ ] I have applied the required changes in production
    - [ ] I cannot apply the required changes in production before this is deployed.

## Description of changes
Adds API for deleting/ removing user data. Put request requires username and password.

### CODE REVIEW

File: apps/authentication/serializers.py [hunk 1]
```diff
@@ -66,6 +71,91 @@ class Meta:
...
+        # Other
+        instance.allergies = None
+        instance.rfid = None
+        instance.nickname = None
+        instance.website = None
+        instance.github = None
+        instance.linkedin = None
+        instance.gender = "male"
+        instance.bio = ""
+        # saldo = serializers.PositiveSmallIntegerField()
```
[the remark this answers is not in the document]
Comment (MrHanSan) [reply apps/authentication/serializers.py]: Agree

**Comment by MrHanSan** [comment]
The G-suite sync appears to be run when the related email objects are deleted. The rest of the related objects gets deleted now.

**Comment by oleast** [comment]
There may still be one thing to fix, at least I can't see that it is removed?:

> The user has a related AllowedUsername object, which is actually the registration of that users membership in Online, but named strangely. This would probably need to be either removed or annulled somehow.

**Comment by MrHanSan** [comment]
Added removal of the AllowedUsername. Also added some randomness to the anonymized username, so reused usernames can be deleted.

**Pull request merged**